### ISSUE TITLE
CA1062 False positive with cast to Nullable

### ISSUE THREAD

**Issue by mdro** [dotnet/roslyn-analyzers#5714]
### Analyzer

**Diagnostic ID**: [CA1062](https://docs.microsoft.com/visualstudio/code-quality/ca1062): `Validate arguments of public methods`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

When casting an `object` parameter to a `Nullable<T>` a false CA1062 triggers on accessing any member of the Nullable

### Steps To Reproduce

```c#
namespace CA1062Repro2
{
    public static class Repro
    {
        public static bool Foo(object? bar)
        {
            int? baz = (int?) bar;
            return baz.HasValue; // false positive (underlining baz); also happens with ToString()
        }
    }
}
```

### Expected behavior

No warning. If `bar` is null the cast results in an empty Nullable.

### Actual behavior

A warning is issued.

## Additional context

This pattern appears in several of our `IValueConverter` implementations because they convert between strings and nullable numbers or dates.
